src/python/pants/option/global_options.py

```diff
@@ -715,7 +771,46 @@ def register_bootstrap_options(cls, register):
             # This default is also hard-coded into the engine's rust code in
             # fs::Store::default_path so that tools using a Store outside of pants
             # are likely to be able to use the same storage location.
-            default=os.path.join(get_pants_cachedir(), "lmdb_store"),
+            default=DEFAULT_LOCAL_STORE_OPTIONS.store_dir,
+        )
+        register(
+            "--local-store-processes-max-size-bytes",
+            type=int,
+            advanced=True,
+            help=(
+                "The maximum size in bytes of the local store containing process cache entries. "
+                f"Stored below `{local_store_dir_flag}`."
+            ),
+            default=DEFAULT_LOCAL_STORE_OPTIONS.processes_max_size_bytes,
+        )
+        register(
+            "--local-store-files-max-size-bytes",
+            type=int,
+            advanced=True,
+            help=(
+                "The maximum size in bytes of the local store containing files. "
+                f"Stored below `{local_store_dir_flag}`."
+                "\n\n"
+                "NB: This size value bounds the total size of all files, but (due to sharding of the "
+                "store on disk) it also bounds the per-file size to (VALUE / 16)."
```

Comment:
This is admittedly pretty frustrating. Another alternative (perhaps for a followup) would be to make the shard count configurable as well (which would be prerequisite for re-benchmarking to ensure that 16 shards are actually necessary: they likely aren't on SSDs).

Reply:
Did you have to configure your mac to get core dumps enabled? If so, then having the default value influenced by a configuration unlikley to be had by the majority? of Pants users seems off. If the default you have is still the right one, can you provide a better justification or else demote the current one to a just-so-happens?

Reply:
I did... and good point. I've increased this further.